app/models/firm.rb

```diff
@@ -118,6 +148,16 @@ def field_order
 
   private
 
+  def geocode_in_progress?
```

Comment:
Obviously we could drop this if you go with the prior comment's recommendations.